test/Swashbuckle.AspNetCore.SwaggerGen.Test/SwaggerGeneratorVerifyTests/SwaggerGeneratorVerifyTests.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.TestSupport;
+using VerifyXunit;
+using Xunit;
+
+namespace Swashbuckle.AspNetCore.SwaggerGen.Test
+{
+    public class SwaggerGeneratorVerifyTests
+    {
+        [Fact]
+        public Task ApiDescriptionsWithMatchingGroupName()
+        {
+            var subject = Subject(
+                apiDescriptions: new[]
+                {
+                    ApiDescriptionFactory.Create<FakeController>(
+                        c => nameof(c.ActionWithNoParameters), groupName: "v1", httpMethod: "POST", relativePath: "resource"),
+
+                    ApiDescriptionFactory.Create<FakeController>(
+                        c => nameof(c.ActionWithNoParameters), groupName: "v1", httpMethod: "GET", relativePath: "resource"),
+
+                    ApiDescriptionFactory.Create<FakeController>(
+                        c => nameof(c.ActionWithNoParameters), groupName: "v2", httpMethod: "POST", relativePath: "resource"),
+                },
+                options: new SwaggerGeneratorOptions
+                {
+                    SwaggerDocs = new Dictionary<string, OpenApiInfo>
+                    {
+                        ["v1"] = new OpenApiInfo { Version = "V1", Title = "Test API" }
+                    }
+                }
+            );
+
+            var document = subject.GetSwagger("v1");
+
+            return Verifier.Verify(document);
```

Comment:
I've only briefly played with Verify before - what's the developer experience like locally and in CI if a test fails, or if a snapshot needs regenerating?

Reply:
Locally it provides a positive developer experience. If the received file doesn't match the snapshot then it opens both files in your diff tool automatically when the test fails. If you want to keep the new received file as the snapshot you can just copy and paste it from the received file to the verified file there and re-run the test. You need to do this every time you add a new test as the verified file will be empty. 

I haven't checked on CI yet, I will push a change to cause a failure to this PR to check.

Reply:
Both files are shown if there's a failure on CI:

```
NotEqual:
  - Received: SwaggerGeneratorVerifyTests.ApiDescriptionsWithMatchingGroupName.DotNet8_0.received.txt
    Verified: SwaggerGeneratorVerifyTests.ApiDescriptionsWithMatchingGroupName.verified.txt

FileContent:

NotEqual:

Received: SwaggerGeneratorVerifyTests.ApiDescriptionsWithMatchingGroupName.DotNet8_0.received.txt
{
  Info: {
    Title: Test API,
    Version: V1
  },
  Paths: {
    /resource: {
      Operations: {
        Get: {
          Tags: [
            {
              Name: Fake,
              UnresolvedReference: false
            }
          ],
          Responses: {
            200: {
              Description: OK,
              UnresolvedReference: false
            }
          },
          Deprecated: false
        },
        Post: {
          Tags: [
            {
              Name: Fake,
              UnresolvedReference: false
            }
          ],
          Responses: {
            200: {
              Description: OK,
              UnresolvedReference: false
            }
          },
          Deprecated: false
        }
      },
      UnresolvedReference: false
    }
  },
  Components: {},
  HashCode: 0F1036F23326E2BD7543491DEC0E03ABCA5BC97AF845BEE30C164155D616F306F6F6E21A77507DA6BE2A85DF1A17B1F07AE23E7608E37C971BAF1CBA2491E899
}
Verified: SwaggerGeneratorVerifyTests.ApiDescriptionsWithMatchingGroupName.verified.txt
{
  Info: {
    Title: Test API,
    Version: V1
  },
  Paths: {
    /resource: {
      Operations: {
        Get: {
          Tags: [
            {
              Name: Fake,
              UnresolvedReference: false
            }
          ],
          Responses: {
            200: {
              Description: OK,
              UnresolvedReference: false
            }
          },
          Deprecated: false
        },
        Post: {
          Tags: [
            {
              Name: Fake,
              UnresolvedReference: false
            }
          ],
          Responses: {
            200: {
              Description: OK
            }
          },
          Deprecated: false
        }
      },
      UnresolvedReference: false
    }
  },
  Components: {},
  HashCode: 0F1036F23326E2BD7543491DEC0E03ABCA5BC97AF845BEE30C164155D616F306F6F6E21A77507DA6BE2A85DF1A17B1F07AE23E7608E37C971BAF1CBA2491E899
}
```